runtime/src/iree/modules/hal/module.c

```diff
@@ -32,97 +32,143 @@
 // Module type definitions
 //===----------------------------------------------------------------------===//
 
-#define IREE_HAL_MODULE_VERSION_0_1 0x00000001u
-#define IREE_HAL_MODULE_VERSION_LATEST IREE_HAL_MODULE_VERSION_0_1
+#define IREE_HAL_MODULE_VERSION_0_2 0x00000002u
+#define IREE_HAL_MODULE_VERSION_LATEST IREE_HAL_MODULE_VERSION_0_2
```

Comment:
> This bumps the HAL version as the existing hal.ex.shared_device op is removed. The runtime HAL module still only reports a single device and the compiler code is still selecting the "default" device (0). Future changes will support multiple `--device=` flags to register multiple available devices in tools and add attributes/lookup for mapping stream/device affinity to device ordinals.

Do you expect this version to change a few more times as part of this work? If so, would it make sense to either develop the changes on a branch or to plan releases around these changes (e.g. push a stable release before this lands, then another after a few patches land).

Reply:
I don't believe so - I think with count/get and the query call we can do everything we need - query takes strings and we can easily add new queries without changing the HAL API itself.